staging/src/k8s.io/apiserver/pkg/server/openapi/openapi_aggregator.go

```diff
@@ -0,0 +1,462 @@
+	}
+	return true
+}
+
+func equalFloatPointer(s1, s2 *float64) bool {
+	if s1 == nil {
+		return s2 == nil
+	}
+	return *s1 == *s2
+}
+
+func equalIntPointer(s1, s2 *int64) bool {
+	if s1 == nil {
+		return s2 == nil
+	}
+	return *s1 == *s2
+}
+
+// EqualSchema returns true if models have the same properties and references
+// even if they have different documentation.
+func EqualSchema(s1, s2 *spec.Schema) bool {
+	if s1 == nil {
+		return s2 == nil
+	}
+	if s1.Ref.String() != s2.Ref.String() {
+		return false
+	}
+	if !equalSchemaMap(s1.Definitions, s2.Definitions) {
+		return false
+	}
+	if !equalSchemaMap(s1.Properties, s2.Properties) {
+		fmt.Println("Not equal props")
+		return false
+	}
+	if !equalSchemaMap(s1.PatternProperties, s2.PatternProperties) {
+		return false
+	}
+	if !equalSchemaArray(s1.AllOf, s2.AllOf) {
+		return false
+	}
+	if !equalSchemaArray(s1.AnyOf, s2.AnyOf) {
+		return false
+	}
+	if !equalSchemaArray(s1.OneOf, s2.OneOf) {
+		return false
+	}
+	if !EqualSchema(s1.Not, s2.Not) {
+		return false
+	}
+	if !equalSchemaOrBool(s1.AdditionalProperties, s2.AdditionalProperties) {
+		return false
+	}
+	if !equalSchemaOrBool(s1.AdditionalItems, s2.AdditionalItems) {
+		return false
+	}
+	if !equalSchemaOrArray(s1.Items, s2.Items) {
+		return false
+	}
+	if !equalStringArray(s1.Type, s2.Type) {
+		return false
+	}
+	if s1.Format != s2.Format {
+		return false
+	}
+	if !equalFloatPointer(s1.Minimum, s2.Minimum) {
+		return false
+	}
+	if !equalFloatPointer(s1.Maximum, s2.Maximum) {
+		return false
+	}
+	if s1.ExclusiveMaximum != s2.ExclusiveMaximum {
+		return false
+	}
+	if s1.ExclusiveMinimum != s2.ExclusiveMinimum {
+		return false
+	}
+	if !equalFloatPointer(s1.MultipleOf, s2.MultipleOf) {
+		return false
+	}
+	if !equalIntPointer(s1.MaxLength, s2.MaxLength) {
+		return false
+	}
+	if !equalIntPointer(s1.MinLength, s2.MinLength) {
+		return false
+	}
+	if !equalIntPointer(s1.MaxItems, s2.MaxItems) {
+		return false
+	}
+	if !equalIntPointer(s1.MinItems, s2.MinItems) {
+		return false
+	}
+	if s1.Pattern != s2.Pattern {
+		return false
+	}
+	if s1.UniqueItems != s2.UniqueItems {
+		return false
+	}
+	if !equalIntPointer(s1.MaxProperties, s2.MaxProperties) {
+		return false
+	}
+	if !equalIntPointer(s1.MinProperties, s2.MinProperties) {
+		return false
+	}
+	if !equalStringArray(s1.Required, s2.Required) {
+		return false
+	}
+	return len(s1.Enum) == 0 && len(s2.Enum) == 0 && len(s1.Dependencies) == 0 && len(s2.Dependencies) == 0
+}
+
+func renameDefinition(s *spec.Swagger, old, new string) {
+	old_ref := DEFINITION_PREFIX + old
+	new_ref := DEFINITION_PREFIX + new
+	newWalkAllRef(func(ref spec.Ref) spec.Ref {
+		if ref.String() == old_ref {
+			return spec.MustCreateRef(new_ref)
+		}
+		return ref
+	}, s).Start()
+	s.Definitions[new] = s.Definitions[old]
+	delete(s.Definitions, old)
+}
+
+// Copy paths and definitions from source to target, rename definitions if needed.
+func MergeSpecs(target, source *spec.Swagger) error {
```

Comment:
"dest" is less ambiguous than "target" :)

I take it this mutates target but not source? maybe state that in the doc string?